### PR TITLE
Add index parser

### PR DESCRIPTION
**Pull request by addaleax**
Take care of #15. @jviotti, please take a look! :) You don’t have to dig through the implementation (did the liblzma work in [my xz fork](https://github.com/addaleax/xz/tree/move-index-parsing-to-liblzma)), but I’d definitely like to know if you’re okay with the way this is laid out in the README.

Left to do:
- [x] Test the various kinds of necessary error handling
- [x] Make CI pass everywhere
- [ ] Suggest `lzma_parse_indexes_from_file` to upstream liblzma

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -475,6 +479,75 @@ Example usage:
...
+
+Read .xz file metadata.
+
+`options.fileSize` needs to be an integer indicating the size of the file
+being inspected, e.g. obtained by `fs.stat()`.
+
+`options.read(count, offset, cb)` must be a function that reads `count` bytes
+from the underlying file, starting at position `offset`. If that is not
+possible, e.g. because the file does not have enough bytes, the file should
+be considered corrupt. On success, `cb` should be called with a `Buffer`
```
Comment (jviotti) [inline README.md]: How does the implementation handle errors sent to `cb`? Does it compare that they are an instance of `Error`? Might be nicer to accept an `Error` or `null` as the first argument, and the `Buffer` as the second one.

**Comment by jviotti** [comment]
Very exciting stuff!

File: README.md (hunk 1 above)
Comment (addaleax) [inline README.md]: I thought of that, but if there’s an I/O error, you have to call handle it yourself anyway, don’t you? As in, what would the parser be supposed to do with that?

Comment (jviotti) [inline README.md]: I think the most obvious behaviour would be for the parser to stop and yield back the same error to the final callback. The gives the client two possibilities:

- Handle the euro themselves, and possibly retry, or whatever.
- Delegate the error to the parser to let it stop.

Comment (addaleax) [inline README.md]: Yeah, makes sense. :+1: It really won’t do anything else beside passing the error to the original callback, though.

Comment (addaleax) [inline README.md]: Published `1.3.1` with the updated behaviour.

Comment (jviotti) [inline README.md]: Awesome, thanks for the great support to get this feature!

Comment (addaleax) [inline README.md]: My pleasure! :) Can’t promise when I’ll get to debug those Windows issues, though…

**Pull request merged**